seed/utils/search.py

```diff
@@ -429,10 +429,24 @@ def build_view_filters_and_sorts(filters: QueryDict, columns: list[dict]) -> tup
     new_filters = Q()
     annotations = {}
     for filter_expression, filter_value in filters.items():
+        parsed_filters, parsed_annotations = _parse_view_filter(filter_expression, filter_value, columns_by_name)
+
+        # if not "", exclude "" and null
         if filter_expression[-4:] == '__ne' and filter_value == '':
             filter_expression = filter_expression.replace('__ne', '__isnull')
-            filter_value = False
-        parsed_filters, parsed_annotations = _parse_view_filter(filter_expression, filter_value, columns_by_name)
+            filter_value = ""  # evals to false
+            is_not_null_parsed_filters, _ = _parse_view_filter(filter_expression, filter_value, columns_by_name)
+
+            parsed_filters &= is_not_null_parsed_filters
+
+        # if exactly "", only return "" and null
+        elif filter_expression[-7:] == '__exact' and filter_value == '':
```

Comment:
It would probably be cleaner to use `.endswith()` for both conditions rather than specific substring lengths